lametro/management/commands/generate_tag_analytics.py

```diff
@@ -1,34 +1,97 @@
 import csv
+from io import StringIO
+from datetime import datetime
 from tqdm import tqdm
+
+from django.conf import settings
 from django.core.management.base import BaseCommand
 
+from googleapiclient.http import MediaIoBaseUpload
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from google.oauth2 import service_account
+
 from lametro.models import LAMetroBill
 
 
 class Command(BaseCommand):
 
-    help = "This command produces a CSV file that lists each Board Report's tags."
+    help = "This command produces a CSV file that lists each Board Report's tags and \
+        uploads it to the 'LA Metro Reports' folder in Google Drive."
 
     def handle(self, *args, **options):
 
-        with open('tag_analytics.csv', 'w', newline='') as f:
-            writer = csv.writer(f, delimiter=',')
-            writer.writerow(
-                ('File ID',
-                 'Identifier'
-                 'Title',
-                 'Last Action Date',
-                 'Tag Classification',
-                 'Tag')
-            )
-
-            for bill in tqdm(LAMetroBill.objects.all()):
-                for tag in bill.rich_topics:
-                    writer.writerow(
-                        (bill.board_report.id,
-                         bill.identifier,
-                         bill.friendly_name,
-                         bill.last_action_date,
-                         tag.classification,
-                         tag.name)
-                    )
+        csv_string = self.generate_tag_analytics()
+        date = datetime.today().strftime('%m_%d_%y')
+        output_file_name = f'{date}_tag_analytics.csv'
+
+
+        file_metadata = {
+            'name': output_file_name,
+            'parents': [settings.REMOTE_ANALYTICS_FOLDER]
+        }
+        drive_service = self.get_google_drive()
+
+        if not drive_service:
+            return
+
+        media = MediaIoBaseUpload(csv_string, mimetype='text/csv')
+        drive_service.files().create(
+                body=file_metadata,
+                media_body=media,
+                fields='id'
+            ).execute()
+
+        self.stdout.write(
+            self.style.SUCCESS(f'Successfully uploaded {output_file_name}!')
+        )
+
+    def generate_tag_analytics(self):
+        """Returns a CSV-formatted byte stream of the tag analytics"""
+
+        csv_string = StringIO()
+        writer = csv.writer(csv_string, delimiter=',')
+        writer.writerow(
+            ('File ID',
+             'Identifier'
+             'Title',
+             'File Name',
+             'Last Action Date',
+             'Tag Classification',
+             'Tag')
```

Comment:
You have seven headings, but you're only writing 6 fields in your loop before. Strike one of "Title" or "File Name"?

Also, what do you think about ordering tag first, then tag classification?

Reply:
Yeah, I think I got confused while looking at the analytics file. For some reason I thought I was missing a heading for one of the columns.

Tag first works too!